core/src/main/java/com/scalar/db/transaction/consensuscommit/RollbackMutationComposer.java

```diff
@@ -77,26 +80,39 @@ private Put composePut(Operation base, TransactionResult result) throws Executio
     TransactionTableMetadata metadata = tableMetadataManager.getTransactionTableMetadata(base);
     LinkedHashSet<String> beforeImageColumnNames = metadata.getBeforeImageColumnNames();
 
-    Map<String, Value<?>> map = new HashMap<>();
+    List<Column<?>> columns = new ArrayList<>();
     result
-        .getValues()
+        .getColumns()
         .forEach(
             (k, v) -> {
               if (beforeImageColumnNames.contains(k)) {
                 String key = k.substring(Attribute.BEFORE_PREFIX.length());
-                map.put(key, v.copyWith(key));
+                if (key.equals(Attribute.VERSION) && v.getIntValue() == 0) {
+                  columns.add(IntColumn.ofNull(Attribute.VERSION));
```

Comment:
For rollback, we need to change 0 -> null as the opposite side of prepare.

Reply:
I think adding a comment for this would be helpful.